mmv1/third_party/terraform/tests/resource_data_loss_prevention_deidentify_template_test.go

```diff
@@ -506,9 +506,167 @@ resource "google_data_loss_prevention_deidentify_template" "basic" {
+            } 
+            bucket_size = 10
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-bucketing-field"
+        }
+        primitive_transformation {
+          bucketing_config {
+            buckets {
+              min {
+                string_value = "00:00:00"
+              }
+              max {
+                string_value = "11:59:59"
+              }
+              replacement_value {
+                string_value = "AM"
+              }
+            }
+            buckets {
+              min {
+                string_value = "12:00:00"
+              }
+              max {
+                string_value = "23:59:59"
+              }
+              replacement_value {
+                string_value = "PM"
+              }
+            }
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-time-part-field"
+        }
+        primitive_transformation {
+          time_part_config {
+            part_to_extract = "YEAR"
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-crypto-hash-field"
+        }
+        primitive_transformation {
+          crypto_hash_config {
+            crypto_key {
+              transient {
+                name = "beep" # Copy-pasting from existing test that uses this field
+              }
+            }
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-date-shift-field"
+        }
+        primitive_transformation {
+          date_shift_config {
+            upper_bound_days = 30
+            lower_bound_days = -30
+            context {
+              name = "unconditionally-date-shift-field"
+            }
+            crypto_key {
+              transient {
+                name = "beep"
+              }
+            }
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-crypto-deterministic-field"
+        }
+        primitive_transformation {
+          crypto_deterministic_config {
+            crypto_key {
+              transient {
+                name = "beep"
+              }
+            }
+            surrogate_info_type {
+              name = "CREDIT_CARD_NUMBER"
+            }
+            context {
+              name = "unconditionally-crypto-deterministic-field"
+            }
+          }
+        }
+      }
+      field_transformations {
+        fields {
+          name = "unconditionally-replace-dictionary-field"
+        }
+        primitive_transformation {
+          replace_dictionary_config {
+            word_list {
+              words = [
+                "foo",
+                "bar",
+                "baz",
+              ]
+            }
+          }
+        }
+      }
     }
   }
 }
+
+resource "google_kms_crypto_key" "my_key" {
```

Comment:
this should probably use a bootstrapped crypto key instead since they can't be deleted easily - for example, see (code generated from) https://github.com/GoogleCloudPlatform/magic-modules/blob/4bddee41debc30f43db234af3c03638254b8493c/mmv1/products/vertexai/terraform.yaml#L33 - there are also handwritten examples in the codebase

Reply:
Ooh I wasn't aware that this was an option, thanks for highlighting it! I'll get it updated soon

Reply:
Addressed in 6b4992c - I don't have the ability to test these tests 'locally' so pushed the commit to get the tests running ASAP

Reply:
[Tests passed](https://github.com/GoogleCloudPlatform/magic-modules/pull/6886#issuecomment-1381921576) - marking resolved